plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3/TestTrinoS3FileSystemAwsS3.java

```diff
@@ -0,0 +1,314 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.s3;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.LocatedFileStatus;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.RemoteIterator;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.net.URI;
+import java.security.SecureRandom;
+import java.util.ArrayList;
+import java.util.List;
+
+import static io.trino.hadoop.ConfigurationInstantiator.newEmptyConfiguration;
+import static java.lang.Character.MAX_RADIX;
+import static java.lang.Math.abs;
+import static java.lang.Math.min;
+import static java.util.Objects.requireNonNull;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+/**
+ * Tests file system operations on AWS S3 storage.
+ * <p>
+ * Requires AWS credentials, which can be provided any way supported by the DefaultProviderChain
+ * See https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html#credentials-default
+ */
+public class TestTrinoS3FileSystemAwsS3
+{
+    private static final SecureRandom random = new SecureRandom();
+    // The suffix needs to be long enough to "prevent" collisions in practice. The length of 5 was proven not to be long enough
+    private static final int RANDOM_SUFFIX_LENGTH = 10;
+
+    private String bucketName;
+    private String s3Endpoint;
+
+    @BeforeClass
+    public void setup()
+    {
+        bucketName = requireNonNull(System.getenv("S3_BUCKET"), "Environment S3_BUCKET was not set");
+        s3Endpoint = requireNonNull(System.getenv("S3_BUCKET_ENDPOINT"), "Environment S3_BUCKET_ENDPOINT was not set");
+    }
+
+    @Test
+    public void testDeleteRecursivelyObjectPath()
+            throws Exception
+    {
+        String prefix = "test-delete-recursively-object-" + randomSuffix();
+
+        try (TrinoS3FileSystem fs = new TrinoS3FileSystem()) {
+            fs.initialize(new URI("s3://%s/".formatted(bucketName)), defaultS3Configuration());
+
+            try {
+                String filename = "file.txt";
+                fs.createNewFile(new Path("s3://%s/%s".formatted(bucketName, prefix), filename));
+                List<String> paths = new ArrayList<>();
+                RemoteIterator<LocatedFileStatus> files = fs.listFiles(new Path("s3://%s/%s".formatted(bucketName, prefix)), false);
+                while (files.hasNext()) {
```

Comment:
It would remove some repetitive code if we'd extract this while loop in a method that takes as parameter `RemoteIterator<LocatedFileStatus>` and returns `List<String>`. I'd also use LinkedList as implementation instead of ArrayList.